### PR TITLE
Manske/add generic datasource selector

### PR DESCRIPTION
**Pull request by manske**

### CODE REVIEW

File: runtime/mount/python_lib/lplots/widgets/tabular_datasource.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+
+@dataclass(frozen=True, kw_only=True)
+class TabularDatasourcePicker:
+    _key: str
+    _state: DataSourceSelectState
+    _signal: Signal[DataSourceValue]
+
+    @property
+    def value(self) -> Any | None:
```
Comment (rteqs) [inline runtime/mount/python_lib/lplots/widgets/tabular_datasource.py:53]: no need None if its already Any

File: runtime/mount/python_lib/lplots/widgets/tabular_datasource.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
+from dataclasses import dataclass
+from typing import Any, Literal, NotRequired, TypedDict
+
+import latch_data_validation
+import latch_data_validation.data_validation
```
Comment (rteqs) [inline runtime/mount/python_lib/lplots/widgets/tabular_datasource.py:5]: del if you're not using. also stick a todo to validate what they pass in

File: runtime/mount/python_lib/lplots/widgets/tabular_datasource.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            node_id = res.get("node_id")
+            if node_id is None or not isinstance(node_id, str):
+                return None
+            lpath = LPath(f"latch://{node_id}.node")
+
+            if lpath.name().endswith(".csv"):
+                return pd.read_csv(lpath.download())
+            elif lpath.name().endswith(".xlsx"):
+                return pd.read_excel(lpath.download())
+            elif lpath.name().endswith(".tsv"):
```
Comment (rteqs) [inline runtime/mount/python_lib/lplots/widgets/tabular_datasource.py:72]: pull lpath.name() into a variable. also check for None

File: runtime/mount/python_lib/lplots/widgets/tabular_datasource.py
Comment (rteqs) [inline runtime/mount/python_lib/lplots/widgets/tabular_datasource.py:1]: name this `datasource.py` kinda long name to import

File: runtime/mount/python_lib/lplots/widgets/tabular_datasource.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+        if res_type == "registry":
+            table_id = res.get("table_id")
+            if table_id is None:
+                return None
+
+            return Table(id=table_id).get_dataframe()
+
+
+def w_tabular_datasource_picker(
```
Comment (rteqs) [inline runtime/mount/python_lib/lplots/widgets/tabular_datasource.py:96]: should prolly just call this w_dataframe_picker?

Comment (rteqs) [reply runtime/mount/python_lib/lplots/widgets/tabular_datasource.py:96]: i wonder if we can just update w_dataframe_picker to do this. we probably need to adjust `w_select` to take in dictionaries as options/default

File: runtime/mount/python_lib/lplots/widgets/tabular_datasource.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+DataSourceType = Literal["ldata", "dataframe", "registry"]
+
+
+class LDataDataSource(TypedDict):
+    node_id: str
+    type: Literal["ldata"]
+
+
+class DataFrameDataSource(TypedDict):
+    df_id: str
```
Comment (rteqs) [inline runtime/mount/python_lib/lplots/widgets/tabular_datasource.py:24]: probably don't want df_id cz i can do `kek  = pd.read_csv(...)` df_id suggest it takes in `df_{num}`

Comment (manske) [reply runtime/mount/python_lib/lplots/widgets/tabular_datasource.py:24]: Huh?

File: runtime/mount/python_lib/lplots/widgets/tabular_datasource.py (hunk 4 above)
Comment (manske) [reply runtime/mount/python_lib/lplots/widgets/tabular_datasource.py:96]: Huh?

Comment (manske) [reply runtime/mount/python_lib/lplots/widgets/tabular_datasource.py]: I'm just gonna name it w_datasource_picker

**Pull request merged**